src/dos/drive_local.cpp

```diff
@@ -218,6 +218,12 @@ bool localDrive::GetSystemFilename(char *sysName, char const * const dosName) {
 
 // Attempt to delete the file name from our local drive mount
 bool localDrive::FileUnlink(char * name) {
+	if (!FileExists(name)) {
+		DEBUG_LOG_MSG("FS: Skipping removal of %s because it doesn't exist",
+		              name);
+		return true;
+	}
+
```

Comment:
@dreamer , with this initial check added here, it makes the `DOS_SetError(DOSERR_ACCESS_DENIED)` at the end of this function perfectly appropriate.
1.  If this new initial check fails, then we know the file exists. 
2. If all attempts to remove the existing file still fail, then there's something beyond our control blocking it - so an access denied code would be the most appropriate response. 

(And so far everything I've thrown at this has passed testing)